cirq/google/serializable_gate_set.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Support for serializing and deserializing cirq.api.google.v2 protos."""
+
+from typing import cast, Dict, Iterable, Optional, Tuple, Union
+
+from cirq.google import op_deserializer, op_serializer
+
+from cirq import circuits, devices, ops, schedules, value
+
+
+class SerializableGateSet:
```

Comment:
Do we really need all this machinery around the concept of serializing a gate set, instead of just serializing the gate set? I'm worried we're pre-emptively creating abstractions that won't actually be useful in more than one case, and that this abstraction won't be more concise than the direct just-do-it imperative code.

Reply:
Yes we do need it.